controllers/protectedapplicationsview_controller.go

```diff
@@ -0,0 +1,567 @@
+	ctx := context.Background()
+
+	if err := mgr.GetCache().IndexField(ctx, &argov1alpha1.ApplicationSet{},
+		clusterDecisionResourceIndexName,
+		func(o client.Object) []string {
+			appSet, ok := o.(*argov1alpha1.ApplicationSet)
+			if !ok {
+				return nil
+			}
+			var placements []string
+
+			for _, gen := range appSet.Spec.Generators {
+				if gen.ClusterDecisionResource != nil {
+					for key, value := range gen.ClusterDecisionResource.LabelSelector.MatchLabels {
+						if strings.Contains(key, placementLabel) {
+							placements = append(placements, value)
+						}
+					}
+				}
+			}
+
+			return placements
+		}); err != nil {
+		return fmt.Errorf("failed to setup ApplicationSet index: %w", err)
+	}
+
+	if err := mgr.GetCache().IndexField(ctx, &ramenv1alpha1.DRPlacementControl{},
+		placementIndexName,
+		func(o client.Object) []string {
+			drpc, ok := o.(*ramenv1alpha1.DRPlacementControl)
+			if !ok {
+				return nil
+			}
+			return []string{drpc.Spec.PlacementRef.Name}
+		}); err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&multiclusterv1alpha1.ProtectedApplicationView{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&ramenv1alpha1.DRPlacementControl{},
+			handler.EnqueueRequestsFromMapFunc(r.findPAVsForDRPC),
+			builder.WithPredicates(r.drpcStatusChangedPredicate())).
+		Watches(&placementv1beta1.Placement{},
+			handler.EnqueueRequestsFromMapFunc(r.findPAVsForPlacement)).
+		Watches(&placementv1beta1.PlacementDecision{},
+			handler.EnqueueRequestsFromMapFunc(r.findPAVsForPlacementDecision)).
+		Complete(r)
+}
+
+func (r *ProtectedApplicationViewReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Logger.With("PAV", req.Name, "Namespace", req.Namespace)
+
+	pav := &multiclusterv1alpha1.ProtectedApplicationView{}
+	if err := r.Get(ctx, req.NamespacedName, pav); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if !pav.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	drpc := &ramenv1alpha1.DRPlacementControl{}
+	drpcKey := types.NamespacedName{
+		Name:      pav.Spec.DRPCRef.Name,
+		Namespace: pav.Spec.DRPCRef.Namespace,
+	}
+	if err := r.Get(ctx, drpcKey, drpc); err != nil {
+		logger.Error("Failed to get DRPlacementControl", "error", err)
+		return ctrl.Result{RequeueAfter: requeueAfterSeconds}, err
+	}
+	placement := &placementv1beta1.Placement{}
+	placementKey := types.NamespacedName{
+		Name:      drpc.Spec.PlacementRef.Name,
+		Namespace: drpc.Spec.PlacementRef.Namespace,
+	}
+	if err := r.Get(ctx, placementKey, placement); err != nil {
+		logger.Error("Failed to get Placement", "error", err)
+		return ctrl.Result{RequeueAfter: requeueAfterSeconds}, err
+	}
+
+	appType, appRef, subRefs, err := r.findApplicationForDRPC(ctx, drpc, placement)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
We don't need this here anymore. We can call this function inside the update PAV Status